src/Sylius/Component/Core/OrderProcessing/StateResolver.php

```diff
@@ -11,49 +11,67 @@
 
 namespace Sylius\Component\Core\OrderProcessing;
 
+use SM\Factory\FactoryInterface;
+use SM\StateMachine\StateMachineInterface;
 use Sylius\Component\Core\Model\OrderInterface;
 use Sylius\Component\Core\Model\OrderShippingStates;
 use Sylius\Component\Core\Model\PaymentInterface;
 use Sylius\Component\Core\Model\ShipmentInterface;
+use Sylius\Component\Core\OrderPaymentStates;
+use Sylius\Component\Core\OrderPaymentTransitions;
 
 /**
  * @author Paweł Jędrzejewski <pawel@sylius.org>
+ * @author Grzegorz Sadowski <grzegorz.sadowski@lakion.com>
  */
 class StateResolver implements StateResolverInterface
 {
     /**
+     * @var FactoryInterface
+     */
+    private $stateMachineFactory;
+
+    /**
+     * @param FactoryInterface $stateMachineFactory
+     */
+    public function __construct(FactoryInterface $stateMachineFactory)
+    {
+        $this->stateMachineFactory = $stateMachineFactory;
+    }
+
+    /**
      * {@inheritdoc}
      */
     public function resolvePaymentState(OrderInterface $order)
     {
-        $paymentState = PaymentInterface::STATE_NEW;
+        $stateMachine = $this->stateMachineFactory->get($order, OrderPaymentTransitions::GRAPH);
+
+        if (OrderPaymentStates::STATE_PAID === $order->getPaymentState()) {
+            return;
+        }
 
         if ($order->hasPayments()) {
```

Comment:
Why we need this big if? Couldn't we simply add the following code to check the amount?
```
if (0 === $completedPaymentTotal) {
    return;
}
```